### ISSUE TITLE
Tumbr rate limit being hit

### ISSUE THREAD

**Issue by OmgImAlexis** [4pr0n/ripme#415]
Since you bundle an API key it's being used too much, maybe allow the user to supply their own key?

```
{"meta":{"status":429,"msg":"Limit Exceeded"},"response":[]}
```

**Comment by Hrxn** [comment]
Should already, work, try to put your key in `rip.properties`.

**Comment by OmgImAlexis** [comment]
There really needs to be some kind of documentation or just remove the default key as it's obviously being used too much. Tumblr does rate limit depending on IP as well as hits so if they've worked out this key is being bundled with this then it'll just keep getting less and less allowed hits before being limited like this.

**Comment by rautamiekka** [comment]
Is there anything provided by Tumblr could be useful for RipMe ?

**Comment by OmgImAlexis** [comment]
@rautamiekka not too sure I understand your question.

**Comment by rautamiekka** [comment]
^ Tumblr-provided docs on API.

**Comment by OmgImAlexis** [comment]
All that needs to be done is the default key needs to be removed and there should be a box asking for the API key when the user types in `.tumblr.com`. Nothing else can really be done.

**Comment by metaprime** [comment]
Documentation: https://www.tumblr.com/docs/en/api/v2

I was under the impression that this API key was a registered API key for the application, in which case they would expect it to be bundled in an application (I think). I think it makes sense to have a default key and maybe prompt the user to register their own key when they try to use the default one. I think if the settings contain no API key then the default one will be used rather than prompting the user, but I'd have to check on this.

**Comment by OmgImAlexis** [comment]
I've had the same thing happen to my own key before as I run a Chrome extension that had a lot of users. After a while Tumblr just ends up lower the limits further and further. It's even worse for things scraping the API.

Another thing you could do is just use the api v1 without a key.

Example: http://reblogalert.tumblr.com/api/read/json?debug=1&start=0
Docs: https://www.tumblr.com/docs/en/api/v1

**Comment by metaprime** [comment]
I'm guessing there's a reason we stopped using the keyless API v1...

**Comment by OmgImAlexis** [comment]
@metaprime looks like you've always used API v2
https://github.com/4pr0n/ripme/commits/master/src/main/java/com/rarchives/ripme/ripper/rippers/TumblrRipper.java

**Comment by metaprime** [comment]
@OmgImAlexis I guess it's worth investigating the keyless API but something tells me each user would pretty quickly go over the rate limit.

**Comment by OmgImAlexis** [comment]
The rate limit per IP for the AP v1 would be a lot higher than the rate limit for everyone using the API v2.

**Comment by metaprime** [comment]
Fair enough...
Although if someone wants to provide an API v2 key wouldn't that be better?
I acknowledge that these two concerns can coexist.

**Comment by OmgImAlexis** [comment]
Not really, the API v1 was depreciated but Tumblr still prefers that over v2 for scraping. I'll see if I can find the email thread I have with one of their devs but they basically confirmed this for one of my own projects as it causes less stress on their servers.

**Comment by metaprime** [comment]
@OmgImAlexis excellent to know!
